db/model/subject.js

```diff
@@ -342,6 +342,12 @@ module.exports = function subject(seq, dataTypes) {
           );
         }
 
+        // Send event for updation of tags, first delete and then add
```

Comment:
With this block of code as it is here, if tags changed, we will basically send a delete and an add, then if the record is published, we'll send an update, and if it's not published, another delete. 

This is wrong.

Your new code has to be inside of the "isPublished" logic below. Follow all the existing isPublished logic, but if tags were changed, send out a "delete" event followed by an "add" event INSTEAD OF the normal "update" event. If tags were not changed, just send out the normal "update" event as it does now.